src/Core/AdminConsole/OrganizationFeatures/Policies/PolicyRequirements/OrganizationDataOwnershipPolicyRequirement.cs

```diff
@@ -63,10 +88,13 @@ public override OrganizationDataOwnershipPolicyRequirement Create(IEnumerable<Po
         var organizationDataOwnershipState = policyDetails.Any()
             ? OrganizationDataOwnershipState.Enabled
             : OrganizationDataOwnershipState.Disabled;
-        var organizationIdsWithPolicyEnabled = policyDetails.Select(p => p.OrganizationId).ToHashSet();
+
+        var policyDetailsByOrganizationId = policyDetails
```

Comment:
I just noticed this, but maybe this part can be moved into the method. One of the benefits is that the computation will only be done when the method is used, making it more optimized.